File: src/CSharp/701-800/784.LetterCasePermutation.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace PeefyLeetCode.LetterCasePermutation
{
public class Solution {
    public IList<string> LetterCasePermutation(string S) {
        List<string> results = new List<string>();
        results.Add(""); 
        foreach (char x in S){
            if ('0' <= x && x <= '9'){
                for (int i = 0; i < results.Count; i++)
                {
                    results[i] = results[i] + x;
                }
            }
            else{
                List<string> temp = new List<string>();
                for (int i = 0; i < results.Count; i++)
                {
                    temp.Add(results[i] + x);
                    temp.Add(results[i] + (char)(x ^ (1 << 5)));
                }
                results = temp;
            }
        }
        return results;
    }
}
}

File: src/Java/701-800/784.LetterCasePermutation.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;

class Solution {
    public List<String> letterCasePermutation(String S) {
        LinkedList<String> results = new LinkedList<>();
        results.add(""); 
        for (char x : S.toCharArray()){
            if ('0' <= x && x <= '9'){
                for (int i = 0; i < results.size(); i++)
                {
                    results.set(i, results.get(i) + x);
                }
            }
            else{
                LinkedList<String> temp = new LinkedList<>();
                for (int i = 0; i < results.size(); i++)
                {
                    temp.add(results.get(i) + x);
                    temp.add(results.get(i) + (char)(x ^ (1 << 5)));
                }
                results = temp;
            }
        }
        return results;
    }
}

